components/script/dom/webglrenderingcontext.rs

```diff
@@ -2285,6 +2290,75 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
         }
     }
 
+    #[allow(unsafe_code)]
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.6
+    unsafe fn GetFramebufferAttachmentParameter(
+        &self,
+        _cx: *mut JSContext,
+        target: u32,
+        attachment: u32,
+        pname: u32
+    ) -> JSVal {
+        // Check if currently bound framebuffer is non-zero as per spec.
+        if self.bound_framebuffer.get().is_none() {
+            self.webgl_error(InvalidOperation);
+            return NullValue();
+        }
+
+        // Note: commented out stuff is for the WebGL2 standard.
+        let target_matches = match target {
+            //constants::READ_FRAMEBUFFER |
+            //constants::DRAW_FRAMEBUFFER => true,
+            constants::FRAMEBUFFER => true,
+            _ => false
+        };
+        let attachment_matches = match attachment {
+            //constants::MAX_COLOR_ATTACHMENTS ... gl::COLOR_ATTACHMENT0 |
+            //constants::BACK |
+            constants::COLOR_ATTACHMENT0 |
+            constants::DEPTH_STENCIL_ATTACHMENT |
+            constants::DEPTH_ATTACHMENT |
+            constants::STENCIL_ATTACHMENT => true,
+            _ => false,
+        };
+        let pname_matches = match pname {
+            //constants::FRAMEBUFFER_ATTACHMENT_ALPHA_SIZE |
+            //constants::FRAMEBUFFER_ATTACHMENT_BLUE_SIZE |
+            //constants::FRAMEBUFFER_ATTACHMENT_COLOR_ENCODING |
+            //constants::FRAMEBUFFER_ATTACHMENT_COMPONENT_TYPE |
+            //constants::FRAMEBUFFER_ATTACHMENT_DEPTH_SIZE |
+            //constants::FRAMEBUFFER_ATTACHMENT_GREEN_SIZE |
+            //constants::FRAMEBUFFER_ATTACHMENT_RED_SIZE |
+            //constants::FRAMEBUFFER_ATTACHMENT_STENCIL_SIZE |
+            //constants::FRAMEBUFFER_ATTACHMENT_TEXTURE_LAYER |
+            constants::FRAMEBUFFER_ATTACHMENT_OBJECT_NAME |
+            constants::FRAMEBUFFER_ATTACHMENT_OBJECT_TYPE |
```

Comment:
So, according to the spec, when `OBJECT_TYPE` is passed, we need to return a `WebGLTexture` or `WebGLRenderbuffer`, right? Not an integer which happens to be the name of that texture / rb on the GPU.

Reply:
WebGL and GLES specs say slightly different things, for some reason.
According to [GLES spec](https://www.khronos.org/registry/OpenGL/specs/es/2.0/es_full_spec_2.0.pdf#nameddest=section-6.1.3), it returns `one of NONE, TEXTURE, or RENDERBUFFER`, which are `typedef`'ed unsigned integers as can be seen in this table:
https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14